go/cmd/vtctldclient/command/vreplication/migrate/migrate.go

```diff
@@ -0,0 +1,148 @@
+		TabletSelectionPreference: tsp,
+		AllTables:                 createOptions.AllTables,
+		IncludeTables:             createOptions.IncludeTables,
+		ExcludeTables:             createOptions.ExcludeTables,
+		OnDdl:                     createOptions.OnDDL,
+		DeferSecondaryKeys:        createOptions.DeferSecondaryKeys,
+		AutoStart:                 createOptions.AutoStart,
+		StopAfterCopy:             createOptions.StopAfterCopy,
+		NoRoutingRules:            createOptions.NoRoutingRules,
+	}
+
+	_, err := common.GetClient().MigrateCreate(common.GetCommandCtx(), req)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addCreateFlags(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVar(&createOptions.SourceKeyspace, "source-keyspace", "", "Keyspace where the tables are being moved from.")
+	cmd.MarkPersistentFlagRequired("source-keyspace")
+	cmd.Flags().StringVar(&createOptions.SourceTimeZone, "source-time-zone", "", "Specifying this causes any DATETIME fields to be converted from the given time zone into UTC.")
+	cmd.Flags().BoolVar(&createOptions.AllTables, "all-tables", false, "Copy all tables from the source.")
+	cmd.Flags().StringSliceVar(&createOptions.IncludeTables, "tables", nil, "Source tables to copy.")
+	cmd.Flags().StringSliceVar(&createOptions.ExcludeTables, "exclude-tables", nil, "Source tables to exclude from copying.")
+	cmd.Flags().BoolVar(&createOptions.NoRoutingRules, "no-routing-rules", false, "(Advanced) Do not create routing rules while creating the workflow. See the reference documentation for limitations if you use this flag.")
+
+	cmd.Flags().StringVar(&createOptions.MountName, "mount-name", "", "Name external cluster is mounted as.")
+	cmd.Flags().StringSliceVarP(&createOptions.Cells, "cells", "c", nil, "Cells and/or CellAliases to copy table data from.")
+	cmd.Flags().Var((*topoproto.TabletTypeListFlag)(&createOptions.TabletTypes), "tablet-types", "Source tablet types to replicate table data from (e.g. PRIMARY,REPLICA,RDONLY).")
+	cmd.Flags().BoolVar(&createOptions.TabletTypesInPreferenceOrder, "tablet-types-in-preference-order", true, "When performing source tablet selection, look for candidates in the type order as they are listed in the tablet-types flag.")
+	onDDLDefault := binlogdatapb.OnDDLAction_IGNORE.String()
+	cmd.Flags().StringVar(&createOptions.OnDDL, "on-ddl", onDDLDefault, "What to do when DDL is encountered in the VReplication stream. Possible values are IGNORE, STOP, EXEC, and EXEC_IGNORE.")
+	cmd.Flags().BoolVar(&createOptions.DeferSecondaryKeys, "defer-secondary-keys", false, "Defer secondary index creation for a table until after it has been copied.")
+	cmd.Flags().BoolVar(&createOptions.AutoStart, "auto-start", true, "Start the MoveTables workflow after creating it.")
+	cmd.Flags().BoolVar(&createOptions.StopAfterCopy, "stop-after-copy", false, "Stop the MoveTables workflow after it's finished copying the existing rows and before it starts replicating changes.")
+}
```

Comment:
We only need to add the --mount-name flag to the common ones, right? I think we can just do this instead:
```
common.AddCommonCreateFlags(migrate)
cmd.Flags().StringVar(&createOptions.MountName, "mount-name", "", "Name external cluster is mounted as.")
```

And that's it. We're building on the common create flags. And we can use the common values as well.

Reply:
We also don't need this function. The migrate command is a global and we can move those two lines into the main register function:
```
common.AddCommonCreateFlags(migrate)
cmd.Flags().StringVar(&createOptions.MountName, "mount-name", "", "Name external cluster is mounted as.")
```

Reply:
I actually started by just invoking `MoveTablesCreate` and using its flags. Note that in `MoveTables` we use `external-cluster-name` which is only needed by `Migrate`. Once the commands worked I switched to this new structure. While I am happy to reuse it, there are a few flags that are not different:
* `atomic-copy` is MoveTables-only
* `source-shards` is MoveTables-only
* `mount-name` is Migrate-only

Given review comments and decisions with previous vtctldclient commands, I decided to use different grpc request protos, though we do end up reusing the existing MoveTablesCreate function.

Let's talk today and decide on this.

Reply:
I'm not talking about the protos or commands here, just the flags and option variables where we store the values. Why have `common.AddCommonCreateFlags` if we're not going to use it? None of the flags in there are MoveTables specific. We should use that and then build on it for what's specific to the command (this is what you had done previously).

Reply:
Ah, right. Updated to use the common create flags.